src/modules/scraper/adapter/messaging/SuggestedFeesConsumer.ts

```diff
@@ -0,0 +1,73 @@
+import { OnQueueFailed, Process, Processor } from "@nestjs/bull";
+import { Logger } from "@nestjs/common";
+import { Job } from "bull";
+import { InjectRepository } from "@nestjs/typeorm";
+import { Repository } from "typeorm";
+import { DateTime } from "luxon";
+import { utils } from "ethers";
+
+import { SuggestedFeesService } from "../../../suggested-fees/service";
+import { SuggestedFeesQueueMessage, ScraperQueue } from ".";
+import { Deposit } from "../../model/deposit.entity";
+import { AppConfig } from "../../../configuration/configuration.service";
+
+@Processor(ScraperQueue.SuggestedFees)
+export class SuggestedFeesConsumer {
+  private logger = new Logger(SuggestedFeesConsumer.name);
+
+  constructor(
+    private appConfig: AppConfig,
+    private suggestedFeesService: SuggestedFeesService,
+    @InjectRepository(Deposit) private depositRepository: Repository<Deposit>,
+  ) {}
+
+  @Process()
+  private async process(job: Job<SuggestedFeesQueueMessage>) {
+    const { depositId } = job.data;
+
+    const deposit = await this.depositRepository.findOne({
+      where: { id: depositId },
+    });
+
+    if (!deposit) {
+      this.logger.verbose(`${ScraperQueue.SuggestedFees} deposit with id '${depositId}' does not exist in db`);
+      return;
+    }
+
+    if (deposit.suggestedRelayerFeePct) {
+      this.logger.verbose(
+        `${ScraperQueue.SuggestedFees} deposit with id '${depositId}' already has a suggested fees entry`,
+      );
+      return;
+    }
+
+    if (!deposit.depositDate) {
+      throw new Error(`Deposit with id '${depositId}' needs 'depositDate' entry in order to fetch suggested fees`);
+    }
+
+    let suggestedRelayerFeePct: string;
+
+    const diffToNowHours = DateTime.fromJSDate(deposit.depositDate).diffNow().as("hours");
```

Comment:
I don't think I discussed with you adding the Luxon library for managing dates, so 👍  for using it. I'm just curious, where did you find the api of `diffNow` and `as`? I couldn't find it in the docs when I needed it and it is a very nice way to compute date differences.

Reply:
I also use luxon quite a lot personally :D Yea I initially discovered them through Intellisense actually. But you can also find them in the API docs
- https://moment.github.io/luxon/api-docs/index.html#datetimediffnow
- https://moment.github.io/luxon/api-docs/index.html#durationas

Reply:
nicee. Thx 🙌